src/VisualStudio/Core/Def/Implementation/Utilities/BrowserHelper.cs

```diff
@@ -35,29 +37,67 @@ public static bool TryGetUri(string link, out Uri uri)
             return true;
         }
 
-        public static Uri CreateBingQueryUri(Workspace workspace, DiagnosticData diagnostic)
+        public static Uri GetHelpLink(DiagnosticDescriptor descriptor, string language)
+            => GetHelpLink(descriptor.Id, descriptor.GetBingHelpMessage(), language, descriptor.HelpLinkUri);
+
+        public static Uri GetHelpLink(DiagnosticData data)
+            => GetHelpLink(data.Id, data.ENUMessageForBingSearch, data.Language, data.HelpLink);
+
+        private static Uri GetHelpLink(string diagnosticId, string title, string language, string rawHelpLink)
         {
-            var errorCode = diagnostic.Id;
-            var title = diagnostic.ENUMessageForBingSearch;
-            workspace.GetLanguageAndProjectType(diagnostic.ProjectId, out var language, out var projectType);
+            if (string.IsNullOrWhiteSpace(diagnosticId))
+            {
+                return default;
```

Comment:
`default` makes it look like the return type is a struct.

```suggestion
                return null;
```